varnish/src/vsc_wrapper.rs

```diff
@@ -0,0 +1,65 @@
+use std::ops::{Deref, DerefMut};
+use std::ffi::CString;
+use std::mem::size_of;
+use varnish_sys::ffi::{vsc_seg, VRT_VSC_Alloc, VRT_VSC_Destroy};
+
+pub unsafe trait VscMetric {
+    fn get_metadata() -> &'static str;
+}
+
+pub struct Vsc<T: VscMetric> {
+    metric: *mut T,
+    seg: *mut vsc_seg,
+    name: CString,
+}
+
+impl<T: VscMetric> Vsc<T> {
+    pub fn new(module_name: &str, module_prefix: &str) -> Self {
+        let mut seg = std::ptr::null_mut();
+        let name = CString::new(module_name).unwrap();
+        let format = CString::new(module_prefix).unwrap();
```

Comment:
use `expect("...")` to explain why these two could fail... i am guessing neither should be coming from the Varnish users, but only from the vmod authors, otherwise we would need to change this to `-> Result<Self, ...>` - so that all the error cases get properly returned.